### PR TITLE
Button | FE | Rename props and expand options

### PR DESCRIPTION
**Pull request by mikemai2awesome**
## Jira

http://vjira2:8080/browse/BDS-1103
http://vjira2:8080/browse/BDS-1102
http://vjira2:8080/browse/BDS-1101

## Summary

Renamed props within the button component to follow current conventions.

## Details

1. `iconOnly` renamed to `icon_only`.
2. `transform` renamed to `text_transform`.
3. `width` responsive options renamed and expanded: 
    * `full-to-auto@xxsmall`
    * `full-to-auto@xsmall`
    * `full-to-auto@small`
    * I held off on changing this to `display`, that requires more discussions and thoughts.
4. Deprecations are indicated in the twig and schema.
5. Existing PL pages are updated to use the new prop names.

## Deprecations

1. `iconOnly`
    * use `icon_only` instead
2. `transform`
    * use `text_transform` instead
3. `width: full@small`
    * use `full-to-auto@small` instead

## How to test

Run the branch locally and check the button docs. Make sure the schema is right and the demos are displaying the props correctly. Check the twig file for proper deprecations.

### CODE REVIEW

File: docs-site/src/pages/pattern-lab/_patterns/02-components/button/40-button-transform-variations.twig [hunk 1]
```diff
@@ -4,7 +4,7 @@
 {% for transform in schema.properties.transform.enum %}
```
Comment (adamszalapski) [inline docs-site/src/pages/pattern-lab/_patterns/02-components/button/40-button-transform-variations.twig]: Please change `schema.properties.transform.enum` to `schema.properties.text_transform.enum` without this change examples are not showing.

Comment (mikemai2awesome) [reply docs-site/src/pages/pattern-lab/_patterns/02-components/button/40-button-transform-variations.twig]: This is all set. Thanks for the catch.

**Comment by mikemai2awesome** [comment]
@sghoweri 

I would like to resolve the other PR (https://github.com/bolt-design-system/bolt/pull/1121) before merging this.

**Comment by mikemai2awesome** [comment]
@sghoweri 

Conflicts resolved and tests passed.

File: packages/components/bolt-icons/src/index.js [hunk 2]
```diff
@@ -113,4 +113,5 @@ export * from './icons/vision';
 export * from './icons/warning';
 export * from './icons/watch';
 export * from './icons/workforce-intelligence';
+export * from './icons/yeti';
```
Comment (sghoweri) [inline packages/components/bolt-icons/src/index.js]: @mikemai2awesome can you make sure we remove this line + the yeti.js file above + the yeti line in the icon schema?

Comment (adamszalapski) [reply packages/components/bolt-icons/src/index.js]: I think that i now why this is added. If the tests fail the cleaning script is not executed and we have this.

Comment (mikemai2awesome) [reply packages/components/bolt-icons/src/index.js]: yeti is killed.

**Comment by colbytcook** [comment]
Created https://pegadigitalit.atlassian.net/browse/DS-108 to track

**Pull request closed without merging**